server/neptune/gateway/event/root_config_builder.go

```diff
@@ -67,7 +67,7 @@ func (b *RootConfigBuilder) Build(ctx context.Context, repo models.Repo, branch
 
 func (b *RootConfigBuilder) build(ctx context.Context, repo models.Repo, branch string, sha string, fileFetcherOptions github.FileFetcherOptions, installationToken int64) ([]*valid.MergedProjectCfg, error) {
 	// Generate a new filepath location and clone repo into it
-	repoDir, cleanup, err := b.RepoFetcher.Fetch(ctx, repo, branch, sha)
+	repoDir, cleanup, err := b.RepoFetcher.Fetch(ctx, repo, branch, sha, fileFetcherOptions.Sha != "")
```

Comment:
For now, we will only do shallow clones for sha based requests (i.e. push events). This may change, so I figured we can just reuse filefetcher options for the repofetcher too.

Reply:
nit: If we plan to change this in the future, maybe we should leave a comment

Reply:
can we not pass a boolean into this.  a boolean can literally only be used for one specific usecase so each time you need to add support for something our interface will just get bigger.

I'd rather just create another options struct which lets us add to it if necessary.  You can populate that one here as well.